sematic/api/endpoints/runs.py

```diff
@@ -494,12 +496,18 @@ def save_graph_endpoint(user: Optional[User]):
         if user is not None:
             run.user_id = user.id
 
+    run_ids = [run.id for run in runs]
+    new_runs = [run for run in runs if run.id not in get_existing_run_ids(run_ids)]
```

Comment:
```suggestion
    existing_run_ids = get_existing_run_ids(run_ids)
    new_runs = [run for run in runs if run.id not in existing_run_ids]
```
otherwise `get_existing_run_ids` would get called for each run